File: src/crtp/crtp/logic/console_logic.py
from crtp.packers.console_packer import ConsolePacker

class ConsoleLogic:

    def __init__(self, CrtpPacker, CrtpLink):
        self.link = CrtpLink
        self.packer = ConsolePacker(CrtpPacker)

    def send_consolepacket(self):
        packet = self.packer.consolepacket()
        self.link.send_packet_no_response(packet)

File: src/crtp_cpp/crtp_logic/console_logic.cpp
#include "crtp_logic/console_logic.hpp"

ConsoleLogic::ConsoleLogic(CrtpLink* crtp_link)
    : Logic(crtp_link),
      packer(ConsolePacker()) {}

void ConsoleLogic::send_consolepacket() {
    auto packet = packer.consolepacket();
    link->send_packet_no_response(&packet);
}